datafusion/core/src/execution/context/mod.rs

```diff
@@ -811,8 +811,14 @@ impl SessionContext {
     /// - `SELECT MY_FUNC(x)...` will look for a function named `"my_func"`
     /// - `SELECT "my_FUNC"(x)` will look for a function named `"my_FUNC"`
     pub fn register_udf(&self, f: ScalarUDF) {
-        self.state
-            .write()
+        let mut state = self.state.write();
+        let aliases = f.aliases();
+        for alias in aliases {
+            state
+                .scalar_functions
+                .insert(alias.to_string(), Arc::new(f.clone()));
```

Comment:
How do we want to handle the fact that another function could be registered with the same alias?  I know this function doesn't return an error, but I suspect that overwriting the alias is not the expected behavior

Reply:
This is an excellent point.

At the moment, the code currently ignores any existing functions that were registered with the same name so I don't think this PR makes that worse (though it is not good)

For this PR perhaps we can just add some documentation that explains  that "any functions registered with the udf name or its aliases are overwritten with this new function"

Reply:
comments added ✅